src/node_http2.cc

```diff
@@ -2178,8 +2344,14 @@ ssize_t Http2Stream::Provider::Stream::OnRead(nghttp2_session* handle,
   Http2Session* session = static_cast<Http2Session*>(user_data);
   DEBUG_HTTP2SESSION2(session, "reading outbound data for stream %d", id);
   Http2Stream* stream = GetStream(session, id, source);
-  if (stream->statistics_.first_byte_sent == 0)
+  if (stream->statistics_.first_byte_sent == 0) {
     stream->statistics_.first_byte_sent = uv_hrtime();
+    TRACE_EVENT_NESTABLE_ASYNC_BEGIN1(
+        NODE_HTTP2_TRACING,
+        "Send Data",
```

Comment:
this says "Send Data", but it is in the `OnRead` method.  Is this correct, or should it be "Receive Data"?

Reply:
The `OnRead` is a bit of a misnomer here. This is the function where nghttp reads data from the *outbound* buffer to *send* to the connected peer, so "Send Data" is correct here.  Again, however, we need to decide what is going to be the most useful bit of data here. Currently, this just gives us the time when nghttp2 begins sending data frames and the moment it stops. What it doesn't tell us is how many chunks of data the usercode actually wrote via the streams API, the latency between those, the size of actual data frames, the latency between data frames, the relationship of that to flow control, and so forth. All of those things are likely going to be far more useful for performance tuning so it's likely that this currently is no where near the level of granularity we need.